internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -371,7 +370,9 @@ func (r *Reconciler) clusterClassToCluster(ctx context.Context, o client.Object)
 	// create a request for each of the clusters.
 	requests := []ctrl.Request{}
 	for i := range clusterList.Items {
-		requests = append(requests, ctrl.Request{NamespacedName: util.ObjectKey(&clusterList.Items[i])})
+		if clusterList.Items[i].GetInfrastructureNamespace() == clusterClass.Namespace {
```

Comment:
Can we add a unit test ensuring that a change in a CC with a given name triggers reconcile only for clusters using it. (and not for cluster using a CC with the same name, but in another ns)
Possibly, this should cover both cluster with ClassNamespace set and not.

Reply:
Note for self: this seems still open no matter if it reports outdated